src/spikeinterface/extractors/neuropixels_utils.py

```diff
@@ -1,13 +1,51 @@
 from __future__ import annotations
 
+import warnings
 import numpy as np
 from typing import Optional
 
+from probeinterface import Probe
+
+
+def get_neuropixels_sample_shifts_from_probe(probe: Probe, stream_name: str = "ap") -> np.ndarray:
+    """
+    Get the inter-sample shifts for Neuropixels probes based on the probe information.
+
+    Parameters
+    ----------
+    probe : Probe
+        The probe object containing channel and ADC information.
+
+    Returns
+    -------
+    sample_shifts : np.ndarray
+        Array of relative phase shifts for each channel.
+    """
+    # get inter-sample shifts based on the probe information and mux channels
+    model_description = probe.annotations.get("model_description", None)
+    num_channels_per_adc = probe.annotations.get("num_channels_per_adc", None)
+    mux_channels = probe.contact_annotations["mux_channels"]
+
+    if "2.0" in model_description:
+        # for Neuropixels 2.0 (and newer), the number of cycles in ADC is equal to the number of channels per ADC
+        num_cycles_in_adc = num_channels_per_adc
+    else:
+        # for Neuropixels 1.0 technology, the number of cycles for the AP stream is +1 because
+        # the last cycle is used for the LFP stream
+        num_cycles_in_adc = num_channels_per_adc + 1 if "ap" in stream_name.lower() else num_channels_per_adc
+    sample_shifts = np.zeros_like(adc_indices)
```

Comment:
```suggestion
    sample_shifts = np.zeros_like(adc_indices, dtype='float')
```
These need to be floats, otherwise Line 39 rounds everything down to 0.

Reply:
float is default, but I'll make it explicit

Reply:
When I tested on our openephys data, `adc_indices` was int, so `sample_shifts` was being initialised as int. Thanks for changing!